sdk/identity/azure-identity/tests/managed-identity-live/service-fabric/service_fabric.md

```diff
@@ -0,0 +1,190 @@
+
+### Create a resource group
+
+From a command prompt window, run
+```
+az login
+az group create -n $RESOURCE_GROUP --location $LOCATION --subscription $SUBSCRIPTION_NAME
+```
+
+### Create a user-assigned managed identity
+
+From your command prompt window, run
+```
+az identity create -g $RESOURCE_GROUP -n AdminUser
+```
+
+Make note of the identity's principal ID, to use in the next step. Also make note of its client ID for a later step.
+
+### Create a key vault, certificate, and secret
+
+Create your key vault:
+```
+az keyvault create -g $RESOURCE_GROUP -n $KEY_VAULT_NAME --sku standard
+```
+
+After creating the vault, create a self-signed certificate in it using the [Azure Portal](https://azure.portal.com/). You'll need to insert some of this certificate's properties into the cluster template later on.
+
+Next, create a secret in the key vault. It can have any name (e.g. "TestSecret") and any value (e.g. "TestValue"). This secret will just be accessed by the Service Fabric applications to verify that they can access a resource and read contents using their identities.
+
+Finally, go to the "Access policies" blade for your vault. Under "Enable Access to:", check the boxes for "Azure Virtual Machines for deployment" and "Azure Resource Manager for template deployment". Be sure to click "Save" at the top of the page. These policies are necessary for your cluster deployment and applications' functionality.
+
+### Create an Azure Container Registry
+
+From your command prompt window, run
+```
+az acr create -g $RESOURCE_GROUP -n $ACR_NAME --admin-enabled --sku basic
+```
+
+### Deploy a managed identity-enabled cluster
+
+At the time of writing, Service Fabric clusters must be deployed using the Azure Resource Manager in order to enable managed identity. Provided is a cluster ARM template that can be used to create a managed identity-enabled cluster once some required fields are completed. The template uses the cluster certificate provided by your key vault, creates a system-assigned identity, and enables the managed identity token service so deployed applications can access their identities.
+
+To use the provided template:
+
+1. Open `arm-templates/cluster.parameters.json` and complete the fields `clusterLocation`, `adminUserName`, `adminPassword`, `sourceVaultValue`, `certificateUrlValue`, and `certificateThumbprint`. Field descriptions will describe how they should be completed.
+2. In `arm-templates/cluster.parameters.json`, change all instances of `sfmi-test` to a unique name, like `<myusername>-sfmi-test`. Also, change the values of `applicationDiagnosticsStorageAccountName` and `supportLogStorageAccountName` to be similarly unique, but without hyphens. This will help ensure the deployment resource names do not conflict with the names of other public resources.
+3. Start the deployment by running from your Powershell window in the `arm-templates` directory:
+```powershell
+Connect-AzAccount
```

Comment:
Using either the CLI or Azure PowerShell would be preferable to using both. You use the CLI more throughout this document; is it possible to deploy the templates with [az deployment](https://docs.microsoft.com/azure/azure-resource-manager/templates/deploy-cli)?